src/Sentry/SentryId.cs

```diff
@@ -31,22 +36,37 @@ namespace Sentry
         /// <returns>String representation of the event id.</returns>
         public override string ToString() => _guid.ToString("n");
 
+        [Conditional("DEBUG")]
+        private void AssertNotEmpty()
+        {
+            Debug.Assert(_guid != Guid.Empty, "Dont use this API with Guid.Empty. Instead use SentryId.Empty");
+        }
+
         /// <inheritdoc />
         public bool Equals(SentryId other) => _guid.Equals(other._guid);
 
         /// <inheritdoc />
         public override bool Equals(object? obj) => obj is SentryId other && Equals(other);
 
         /// <inheritdoc />
-        public override int GetHashCode() => _guid.GetHashCode();
+        public override int GetHashCode()
+        {
+            AssertNotEmpty();
```

Comment:
I believe this would interfere with comparison and equality operations.

Reply:
Did some research and you are correct - it would assert when used in things like `HashSet<SentryId>` or `Dictionary<SentryId, object>`, etc.

I don't want to prevent adding `SentryId.Empty` to a dictionary - there may be a valid use for that now or in the future. We mainly want to avoid sending them to Sentry, not using them internally.

Also, I found: https://learn.microsoft.com/dotnet/api/system.object.gethashcode#notes-to-inheritors
Which says (among other things):

> The `GetHashCode()` method should not throw exceptions.

And a few various StackOverflow posts about that as well.

Let's leave it out of `GetHashCode`.